pkg/config/process.go

```diff
@@ -0,0 +1,61 @@
+package config
+
+import (
+	"os"
+	"time"
+)
+
+const (
+	// DefaultGRPCConnectionTimeoutSecs sets the default value for timeout when connecting to the agent
+	DefaultGRPCConnectionTimeoutSecs = 60
+)
+
+func setupProcesses(config Config) {
+	// Note that `BindEnvAndSetDefault` automatically creates environment variables with the `DD_PROCESS_CONFIG` prefix.
+	// This means that we must manually add the same environment variable, but with the `DD_PROCESS_AGENT` prefix if we want to support that too.
+
+	config.SetDefault("process_config.enabled", "false")
+	// process_config.enabled is only used on Windows by the core agent to start the process agent service.
+	// it can be set from file, but not from env. Override it with value from DD_PROCESS_AGENT_ENABLED.
+	ddProcessAgentEnabled, found := os.LookupEnv("DD_PROCESS_AGENT_ENABLED")
+	if found {
+		AddOverride("process_config.enabled", ddProcessAgentEnabled)
+	}
+
+	config.BindEnv("process_config.process_dd_url", "")
+
+	config.SetKnown("process_config.dd_agent_env")
+	config.SetKnown("process_config.enabled")
+	config.SetKnown("process_config.intervals.process_realtime")
+	config.SetKnown("process_config.queue_size")
+	config.SetKnown("process_config.rt_queue_size")
+	config.SetKnown("process_config.max_per_message")
+	config.SetKnown("process_config.max_ctr_procs_per_message")
+	config.SetKnown("process_config.cmd_port")
+	config.SetKnown("process_config.intervals.process")
+	config.SetKnown("process_config.blacklist_patterns")
+	config.SetKnown("process_config.intervals.container")
+	config.SetKnown("process_config.intervals.container_realtime")
+	config.BindEnvAndSetDefault("process_config.dd_agent_bin", DefaultDDAgentBin, "DD_PROCESS_CONFIG_DD_AGENT_BIN", "DD_PROCESS_AGENT_DD_AGENT_BIN")
```

Comment:
aren't the `DD_PROCESS_CONFIG_` env vars automatically generated?

Reply:
They should be, however for some reason having them automatically generated breaks `TestOnlyEnvConfig` I will add a test that checks to see if there’s a corresponding DD_PROCESS_CONFIG setting for every DD_PROCESS_AGENT setting, so we can understand this behavior a little better.

I think this is kind of cosmetic, so it can probably go in without it and we can correct it later if it gets too close to the freeze

Reply:
I did some more investigation into this. Basically, `BindEnvAndSetDefault` will only generate an environment variable if you do not specify one.  If you give no env var (`ex BindEnvAndSetDefault("process_config.foo", "asdf")`), it will bind `DD_PROCESS_CONFIG_FOO` to process_config.foo. However, if you give it a literal env var BindEnvAndSetDefault("process_config.foo", "asdf", "BAR") , it will not generate `DD_PROCESS_CONFIG_FOO` for you, and will only bind `BAR`

Reply:
Oh, that makes sense! Thanks for clarifying it!!